views/cypress/tests/delivery/basic-nonlinear-test.spec.js

```diff
@@ -0,0 +1,56 @@
+/**
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU General Public License
+ * as published by the Free Software Foundation; under version 2
+ * of the License (non-upgradable).
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ *
+ * Copyright (c) 2021 Open Assessment Technologies SA ;
+ */
+
+/**
+ * Note: this test requires the following:
+ * - a dedicated user is created and defined in the the cypress env file through the properties testTakerUser and testTakerPass
+ * - the test `taoQtiTest/views/cypress/fixtures/testPackages/basic-nonlinear-test.zip` is imported
+ * - the test is published and assigned to the user defined in the env file
+ * - the delivery name is supplied through the cypress env file from the property deliveryIds.basicNonLinearTest
+ *
+ * See the fixtures folder
+ */
+
+import { basicNonLinearFirstLaunchSpecs, basicNonLinearSecondLaunchSpecs } from './shared/basic-nonlinear-test';
```

Comment:
@jsconan  Could you explain the goal of this extension? It is not very clear from read.me, but for me it is strange to see shared tests here, but not in 'tao-core'.

Reply:
Looks like it's because test content is copied in another repository https://github.com/oat-sa/extension-tao-ltideliveryprovider/blob/master/views/cypress/tests/delivery/basic-linear-test.spec.js#L30 (2 ways to launch are tested)

Reply:
I understand this, but could we use tests from "extension-tao-testqti" directly in "extension-tao-ltideliveryprovider" without duplicating code in both the extensions? For example point in cypress config [path to test folder](https://docs.cypress.io/guides/references/configuration#Folders-Files)?

Reply:
tao core must remain agnostic. We cannot add things that are specific to another extension in this package.

Then, regarding the test being placed in another module, yes this is for the same reason. In taoQtiTest we can only rely on the regular way to launch a test. The LTI mode is specific to the ltiDeliveryProvider extension. Instead of duplicating the same test, it is better to share it with respect to the dependencies order.